src/shogun/classifier/Perceptron.cpp

```diff
@@ -17,56 +17,39 @@
 
 using namespace shogun;
 
-CPerceptron::CPerceptron()
-: CLinearMachine()
+CPerceptron::CPerceptron() : CIterativeLinearMachine()
 {
 	init();
```

Comment:
I think since we only have a single (default) constructor, there is no need for this init method, but rather copy its contents into the ctor directly